generator/codegen/collection_extensions.go

```diff
@@ -0,0 +1,288 @@
+package codegen
+
+import (
+	"encoding/json"
+	"fmt"
+	"slices"
+	"strings"
+
+	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/getkin/kin-openapi/openapi3"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+)
+
+const (
+	schemaPrefix        = "#/components/schemas/"
+	jsonMIME            = "application/json"
+	collectionFlag      = "x-collection"
+	notificationFeedRef = "NotificationFeed"
```

Comment:
could you add a comment mentioning the fact that messages are a special case : a feed rather than just a collection